oemoflex/tools/plots.py

```diff
@@ -0,0 +1,144 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
+import yaml
+from matplotlib.ticker import EngFormatter
+
+from pandas.plotting import register_matplotlib_converters
+register_matplotlib_converters()
+
+
+def map_labels(df, labels_dict, bus_name, demand_name):
+    r"""
+    Changes the column names and makes electricity consumers negative except demand.
+
+    Parameters
+    ---------------
+    'df' : pandas.DataFrame
+        Dataframe with electricity data.
+    'labels_dict' : dictionary
+        Contains old and new column names. The new column names are used for the labels in the plot.
+    'bus_name' : string
+        Name of the bus to identify columns where electricty goes from the bus to a consumer.
+    'demand_name' : string
+        Name of the demand bus to identify the column where demand is the consumer.
+
+    Returns
+    ----------
+    'df' : pandas.DataFrame
+        Edited dataframe with new column names and negative sign for consumer columns.
+    """
+    df.columns = data.columns.to_flat_index()
+    for i in df.columns:
+        if i[0] == bus_name and not (i[1] == demand_name):
+            df[i] = df[i] * -1
+            df.rename(columns={i: labels_dict[i]}, inplace=True)
+        else:
+            df.rename(columns={i: labels_dict[i]}, inplace=True)
+
+    return df
+
+def filter_timeseries(df, timestamp_col, start_date, end_date):
```

Comment:
Instead of passing the timeindex as a column, you can demand that it is set as the index of the dataframe.

Maybe you can even assert that the index is a timeindex: `assert isinstance(df.index, pd.DatetimeIndex)`

Reply:
I included the assert command.
Can I do more to demand that the timeindex is set as the index of the dataframe?